### PR TITLE
[Designer] Add undo/redo successful popup for narrator

### LINKED ISSUE #7642: [AdaptiveCards- Designer - Host App Docs]: After performing the 'Undo or Redo' action, screen reader does not provide any conformation.
### Target Platforms

Other

### SDK Version

1.5

### Application Name

Adaptive Cards

### Problem Description

Test Environment:

OS: Version 21H2 (OS Build 22000.318)

URL: [https://adaptivecards.io/](https://nam06.safelinks.protection.outlook.com/?url=https://adaptivecards.io/&data=04%7c01%7cv-ibalam%40microsoft.com%7cbd35d875ec3f4539762308d88f709775%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637417061036711036%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=dUolQbKGi1J8gsoxC2JG5TGoNi41msPfIIjKSIyPGHM%3D&reserved=0)

Edge Dev: Version 105.0.1300.0 (Official build) dev (64-bit)

Screen Reader: Narrator

Repro Steps:

1. Open above URL with valid credential.

2. Navigate to 'Designer' link and press enter key to activate it.

3. Navigate to 'Redo' button and press tab key.

4. Navigate to 'Undo or Redo' button and press enter key to activate it and observe the issue.

Actual Result:

After performing the 'Undo or Redo' action, screen reader is announcing as Undo/Redo disabled.

Expected Result:

when Undo/Redo action is performed, Screen Reader should convey the information as "undo your last changes and undo disabled" .

Note:

Same issue is repro with NVDA as well.

User Experience:

It will impact the screen reader user if the screen reader does not convey the message to the user.

MAS Reference:
https://liquid.microsoft.com/Web/Object/Read/MS.Accessibility/Requirements/01.03.01

"Have feedback to share on Bugs ? Please help fill [Trusted Tester Bug Feedback (office.com)](https://nam06.safelinks.protection.outlook.com/?url=https://forms.office.com/pages/responsepage.aspx?id%3Dv4j5cvGGr0GRqy180BHbR5dUqGQgJ9lItv-jqY8F09lURVdYTENIRkY2UzlCR0NKV1E2VUE5OVpDVy4u&data=04%7c01%7cv-apoorva%40microsoft.com%7c551850210d444be9b28308d9691e3dde%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637656400998658942%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=l6w0SvVcMSvyGnJ6%2B6uSSgd3braF2dZ7/AdBrqqzIR8%3D&reserved=0) “.

### Screenshots

https://user-images.githubusercontent.com/104482612/177278840-812d9eb3-5ab4-48e5-a8b6-140afee71903.mp4

### Card JSON

```JSON
NA
```

### Sample Code Language

NA

### Sample Code

NA

### PR DESCRIPTION
**Pull request by anna-dingler**
# Related Issue

Fixes #7642

# Description

Resolves an accessibility issue by adding a popup message to show the confirmation of an undo or redo action.

Old behavior: Narrator only announced that the undo/redo button was disabled

Updated behavior: Narrator announces that the action was successful and that the undo/redo button is disabled.

# How Verified

Verified manually on the Adaptive Cards site.

undoRedo

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/7783)

### CODE REVIEW

**Review by jwoo-msft — APPROVED** [review]
nit: I just want to be certain; the narrator will announce the undo and redo actions are successful, right?
Approved it anyway since the description said so.

**Comment by anna-dingler** [comment]
> 

Yes, Narrator will announce the success popup.

Upon focus of the Undo button, Narrator says "Undo Button. Undo your last change"
Upon clicking the Undo button, Narrator says "Undo disabled. Alert: Successfully undid your last change"

File: source/nodejs/adaptivecards-designer/src/card-designer.ts [hunk 1]
```diff
@@ -1473,6 +1473,15 @@ export class CardDesigner extends Designer.DesignContext {
 
             this.setCardPayload(card, false);
             this.updateToolbar();
+
+            const undoSuccessPopup = document.createElement("p");
+            undoSuccessPopup.textContent = "Successfully undid your last change";
```
Comment (licanhua) [inline source/nodejs/adaptivecards-designer/src/card-designer.ts:1478]: what's message used in other product or other webside? will "Undo Complete" and "Redo Complete" be better?

File: source/nodejs/adaptivecards-designer/src/card-designer.ts [hunk 2]
```diff
@@ -1473,6 +1473,15 @@ export class CardDesigner extends Designer.DesignContext {
...
+            undoSuccessPopup.className = "popup-message";
+            undoSuccessPopup.setAttribute("role", "alert");
+
+            document.body.appendChild(undoSuccessPopup);
+
+            setTimeout(() => { undoSuccessPopup.remove(); }, 3000);
```
Comment (licanhua) [inline source/nodejs/adaptivecards-designer/src/card-designer.ts:1484]: Is there accessiblity practice to state how long should a popup shows?

File: source/nodejs/adaptivecards-designer/src/card-designer.ts (hunk 1 above)
Comment (anna-dingler) [reply source/nodejs/adaptivecards-designer/src/card-designer.ts:1478]: I couldn't find an example online, but I do like the shorter text

File: source/nodejs/adaptivecards-designer/src/card-designer.ts (hunk 2 above)
Comment (anna-dingler) [reply source/nodejs/adaptivecards-designer/src/card-designer.ts]: I couldn't find a best practice on this. 

With the shortened text, 3 seconds is long enough for Narrator to fully read it before the popup is dismissed.

**Comment by anna-dingler** [comment]
The a11y team has closed this bug, so I'm closing the PR as well.

**Pull request closed without merging**